portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -14,17 +14,68 @@
 
 package com.google.sps.data;
 
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.helper.ValidationResult;
+import com.google.sps.servlets.DataServlet;
+import javax.servlet.http.HttpServletRequest;
+
 /** Class containing comment data. */
 public final class Comment {
 
+  // Whitelist to avoid the possibility of XSS
+  private static final String UNSAFE_CHARACTERS_REGEX = "[^A-Za-z0-9._~()'!*:@,;+?\\s-]";
+
   public static final String AUTHOR_KEY = "author";
   public static final String BODY_KEY = "comment-body";
   public static final String TIME_POSTED_KEY = "timePosted";
+  
+  /**
+   * Get and validate a comment object from an incoming request.
+   * @param request
+   * @return Returns a new comment object or a validation error message.
+   */
+  public static ValidationResult<Comment> getIncomingComment(HttpServletRequest request) {
+    String commentAuthor = DataServlet.getParameter(request, AUTHOR_KEY, "");
+    String commentBody = DataServlet.getParameter(request, BODY_KEY, "");
+
+    commentAuthor = commentAuthor.replaceAll(UNSAFE_CHARACTERS_REGEX, "");
```

Comment:
Just a comment, I think I saw earlier you had written about sanitizing the HTML and how innerhtml takes care of some of that.

I generally try to avoid writing my own security code (in the principle of [not rolling your own](https://security.stackexchange.com/questions/18197/why-shouldnt-we-roll-our-own) since I'm not an expert. Maybe I'd take a look at other vetted open source sanitizers like [OWASP](https://github.com/OWASP/java-html-sanitizer) and see if we can include that in our packages?

But this is fine, having 1 extra safety layer is better than nothing.

Reply:
OWASP looks great. I'll look into that more if I have additional time to spare sometime. For now I'm mostly going with the "something is better than nothing" mindset. I also have a rather restrictive whitelist of permitted characters that I would imagine would make it rather difficult for XSS, but again not rolling your own.

Reply:
SG!

Reply:
I'm going to start putting stuff like this into the issues section.